packages/components/src/pages/VsCodeExtension.vue

```diff
@@ -580,7 +583,9 @@ export default {
 
   mounted() {
     this.$root.$on('makeRoot', (codeObject) => {
+      this.clearSelection();
```

Comment:
De-selecting the active node seems like an acceptable solution to the reactivity in the left sidebar. Is it a difficult fix to make the sidebar reactive?

Reply:
To make it reactive `selectedObject` should be changed, but after setting different root `classMap` is changed and it's not possible to re-select active `codeObject`, because ID of object has changed.

For example, if you select `ActiveSupport::SecurityUtils` node, ID of selected object is:
`class:active_support/ActiveSupport::SecurityUtils`
but when user set that object as root ID of that object becomes `SecurityUtils`

Reply:
The issue seems that the panel takes an `AppMap` object as a prop. So that prop will be reactive (i.e., if the AppMap changes the panel will update), however changes to its children will not update the panel. The fix would be to switch the prop to a `CodeObject[]`. Sounds like this could be its own issue.

Reply:
Selected object info (inbound/outbound connections, events, etc.) is taken from `selectedObject`, but it doesn't change when `AppMap` changes.
https://github.com/applandinc/appmap-js/blob/feat/global-set-as-root/packages/components/src/components/DetailsPanel.vue#L28-L32

Reply:
And we can't update `selectedObject` by hand after appmap changes, because object's IDs were changed. @dustinbyrne